crates/core/src/host/wasm_common/instrumentation.rs

```diff
@@ -0,0 +1,69 @@
+use std::time::{Duration, Instant};
```

Comment:
The span stuff is rough and ready, happy to do whatever with it once we agree this is all the right direction.